spec/cohortManager.spec.js

```diff
@@ -0,0 +1,217 @@
+    // execute
+    const res = cm.createCohort('test1')
+    // verfiy
+    expect(res).toEqual(expected)
+  })
+
+  it('find a cohort by name', () => {
+    // set up
+    cm.createCohort('test1')
+    cm.createCohort('test2')
+    cm.createCohort('test3')
+    const expected = new Cohort('test2')
+    // execute
+    const res = cm.searchCohort('test2')
+    // verify
+    expect(res).toEqual(expected)
+  })
+
+  it('return an error message if cohort is not found', () => {
+    // set up
+    cm.createCohort('test1')
+    cm.createCohort('test2')
+    cm.createCohort('test3')
+    const expected = 'Cohort does not exist'
+    // execute
+    const res = cm.searchCohort('test5')
+    // verify
+    expect(res).toEqual(expected)
+  })
+
+  it('create a new Student', () => {
+    // set up
+    const expected = [new Student(1, 'Max', 'Mustermann')]
+    // execute
+    cm.newStudent('Max', 'Mustermann')
+    // verfiy
+    expect(cm.students).toEqual(expected)
+  })
+
+  it('create Multiple students', () => {
+    // set up
+    const expected = [
+      new Student(1, 'Max', 'Mustermann'),
+      new Student(2, 'Ryley', 'Suzanne'),
+      new Student(3, 'Ziya', 'Blagorodna')
+    ]
+    // execute
+    cm.newStudent('Max', 'Mustermann')
+    cm.newStudent('Ryley', 'Suzanne')
+    cm.newStudent('Ziya', 'Blagorodna')
+    // verify
+    expect(cm.students).toEqual(expected)
+  })
+
+  it('add a student to a cohort', () => {
+    // set up
+    const expected = new Cohort('test')
+    const expectedStudent = new Student(1, 'Max', 'Mustermann')
+    expectedStudent.inCohort = true
+    expected.students.push(expectedStudent)
+    cm.createCohort('test')
+    cm.newStudent('Max', 'Mustermann')
+    // execute
+    cm.addStudentToCohort(1, 'test')
+    // verfiy
+    expect(cm.cohorts[0]).toEqual(expected)
+  })
+
+  it('add a student to a cohort', () => {
+    // set up
+    const expected = new Cohort('test2')
+    const expectedStudent = new Student(1, 'Max', 'Mustermann')
+    expectedStudent.inCohort = true
+    expected.students.push(expectedStudent)
+    cm.createCohort('test1')
+    cm.createCohort('test2')
+    cm.createCohort('test3')
+    cm.newStudent('Max', 'Mustermann')
+    // execute
+    cm.addStudentToCohort(1, 'test2')
+    // verfiy
+    expect(cm.cohorts[1]).toEqual(expected)
+  })
+
+  it('return an error message if student does not exist', () => {
+    // set up
+    const expected = 'Student does not exist'
+    cm.createCohort('test')
+    // execute
+    const res = cm.addStudentToCohort(1, 'test')
+    // verfiy
+    expect(res).toEqual(expected)
+  })
+
+  it('return an error message if cohort does not exist', () => {
+    // set up
+    const expected = 'Cohort does not exist'
+    cm.newStudent('Max', 'Mustermann')
+    // execute
+    const res = cm.addStudentToCohort(1, 'test')
+    // verify
+    expect(res).toEqual(expected)
+  })
+
+  it('remove a cohort by name', () => {
```

Comment:
you're not testing the behaviour of removing a cohort that has students:

a) is that even allowed?
b) if it is allowed, are you checking that all students are now marked as not being in a cohort?